ESP8266/ESP8266.cpp

```diff
@@ -404,26 +406,31 @@ nsapi_error_t ESP8266::open_udp(int id, const char* addr, int port, int local_po
 
     _smutex.lock();
     if(local_port) {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port)
-                && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d,%d", id, type, addr, port, local_port);
     } else {
-        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port)
-               && _parser.recv("OK\n");
+        done = _parser.send("AT+CIPSTART=%d,\"%s\",\"%s\",%d", id, type, addr, port);
     }
 
     if (done) {
+        if (!_parser.recv("OK\n")) {
+            if (_sock_already) {
+                _sock_already = false; // To be raised again by OOB msg
+                _smutex.unlock();
+                return NSAPI_ERROR_IS_CONNECTED;
```

Comment:
Is this correct?

If Socket returns `ERROR_IS_CONNECTED` it means that the socket is now connected, and you can start sending to it.

But this is not the case here.